src/marqo/s2_inference/clip_utils.py

```diff
@@ -156,15 +185,195 @@ def __init__(self, model_type: str = "ViT-B/32", device: str = 'cpu',  embedding
         self.processor = None
         self.embedding_dimension = embedding_dim
         self.truncate = truncate
+        self.model_properties = kwargs["model_properties"]
```

Comment:
is there any validation on "model_properties"? it might be better to have a default 
` self.model_properties = kwargs.get("model_properties", dict())` (check the syntax, but something like that)

Reply:
I think this still needs to be implemented?